File: app/src/main/kotlin/com/absinthe/libchecker/utils/elf/source/DataSource.java
package com.absinthe.libchecker.utils.elf.source;

import java.io.Closeable;
import java.io.IOException;
import java.nio.ByteBuffer;
import java.nio.ByteOrder;

public interface DataSource extends Closeable {
    void read(ByteBuffer buffer, long offset) throws IOException;
    void read(long offset, byte[] buf, int length) throws IOException;
    long readWord(long offset) throws IOException;
    int readHalf(long offset) throws IOException;
    long readLong(long offset) throws IOException;
    int readInt(long offset) throws IOException;
    short readByte(long offset) throws IOException;
    String readString(long offset) throws IOException;
    void position(long offset) throws IOException;
    ByteOrder getOrder();
    void setOrder(ByteOrder order);
}

File: app/src/main/kotlin/com/absinthe/libchecker/utils/elf/source/DataSource.kt
package com.absinthe.libchecker.utils.elf.source

import java.io.Closeable
import java.io.IOException
import java.nio.ByteBuffer
import java.nio.ByteOrder

interface DataSource : Closeable {
  @Throws(IOException::class)
  fun read(buffer: ByteBuffer, offset: Long)
  
  @Throws(IOException::class)  
  fun read(offset: Long, buf: ByteArray, length: Int)
  
  @Throws(IOException::class)
  fun readWord(offset: Long): Long
  
  @Throws(IOException::class)
  fun readHalf(offset: Long): Int
  
  @Throws(IOException::class)
  fun readLong(offset: Long): Long
  
  @Throws(IOException::class)
  fun readInt(offset: Long): Int
  
  @Throws(IOException::class)
  fun readByte(offset: Long): Short
  
  @Throws(IOException::class)
  fun readString(offset: Long): String
  
  @Throws(IOException::class)
  fun position(offset: Long)
  
  val order: ByteOrder
  
  fun setOrder(order: ByteOrder)
}